Sources/Utility/ArgumentParser.swift

```diff
@@ -382,12 +361,94 @@ public final class ArgumentParser {
         }
     }
 
+    /// An enum representing way to parse argument values.
+    public enum ParsingStrategy {
+        case one
+        case multiple
+        case remainder
+        
+        func parse(_ parser: inout Parser, kind: ArgumentKind.Type) throws -> [ArgumentKind] {
+            var values: [ArgumentKind] = []
+            
+            switch self {
+            case .one:
+                var parserProtocol = parser as ArgumentParserProtocol
+                try values.append(kind.init(parser: &parserProtocol))
+                parser = parserProtocol as! Parser
+            case .multiple:
+                while true {
+                    let previousIterator = parser.argumentsIterator
+                    
+                    guard let nextArgument = try? parser.next() else {
+                        break
+                    }
+                    
+                    if isPositional(argument: nextArgument) {
+                        try values.append(kind.parse(argument: nextArgument))
+                    } else {
+                        parser.argumentsIterator = previousIterator
+                        break
+                    }
+                }
+            case .remainder:
+                try values.append(contentsOf: parser.argumentsIterator.map({
+                    try kind.parse(argument: $0)
+                }))
+            }
+            
+            return values
+        }
+    }
+
+    /// A struct for the parser representation of option arguments.
+    public struct ParserOptionArgument {
+        let argument: AnyArgument
+        let strategy: ParsingStrategy
+        
+        func update(_ result: Result, with parser: inout Parser) throws {
+            if let associatedArgument = parser.associatedArgumentValue {
+                let value = try argument.kind.parse(argument: associatedArgument)
+                try result.add([value], for: argument)
+                return
```

Comment:
`names` will contain `["foo"]` and it will try to treat `bar` as a positional argument.